geo: fix invalid GeoJSON output

otan opened the pull request:
Resolves #71001

Release note (bug fix): Fixed a panic that could occur with invalid
GeoJSON input.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/71110)
<!-- Reviewable:end -->

otan committed:
geo: fix invalid GeoJSON output

Release note (bug fix): Fixed a panic that could occur with invalid
GeoJSON input using ST_GeomFromGeoJSON/ST_GeogFromGeoJSON.

otan commented:
bors r=rafiss
